src/shmem.c

```diff
@@ -193,24 +193,29 @@ static pthread_mutex_t create_mutex(void) {
 static void remap_shm(void)
 {
 	// Remap shared object pointers which might have changed
-	realloc_shm(&shm_queries, counters->queries_MAX*sizeof(queriesData), false);
+	realloc_shm(&shm_queries, counters->queries_MAX, sizeof(queriesData), false);
 	queries = (queriesData*)shm_queries.ptr;
 
-	realloc_shm(&shm_domains, counters->domains_MAX*sizeof(domainsData), false);
+	realloc_shm(&shm_domains, counters->domains_MAX, sizeof(domainsData), false);
 	domains = (domainsData*)shm_domains.ptr;
 
-	realloc_shm(&shm_clients, counters->clients_MAX*sizeof(clientsData), false);
+	realloc_shm(&shm_clients, counters->clients_MAX, sizeof(clientsData), false);
 	clients = (clientsData*)shm_clients.ptr;
 
-	realloc_shm(&shm_upstreams, counters->upstreams_MAX*sizeof(upstreamsData), false);
+	realloc_shm(&shm_upstreams, counters->upstreams_MAX, sizeof(upstreamsData), false);
 	upstreams = (upstreamsData*)shm_upstreams.ptr;
 
-	realloc_shm(&shm_dns_cache, counters->dns_cache_MAX*sizeof(DNSCacheData), false);
+	realloc_shm(&shm_dns_cache, counters->dns_cache_MAX, sizeof(DNSCacheData), false);
 	dns_cache = (DNSCacheData*)shm_dns_cache.ptr;
 
-	realloc_shm(&shm_strings, counters->strings_MAX, false);
+	realloc_shm(&shm_strings, counters->strings_MAX, sizeof(char), false);
 	// strings are not exposed by a global pointer
 
+	// Total number of regex filters
+	const unsigned int num_regex_tot = counters->num_regex[REGEX_BLACKLIST] +
+	                                   counters->num_regex[REGEX_WHITELIST];
+	realloc_shm(&shm_per_client_regex, counters->clients, num_regex_tot * sizeof(bool), false);
```

Comment:
Adding this line is the actual bugfix, the rest is merely making debug messages more useful in the future.